Makefile

```diff
@@ -15,18 +15,18 @@ $(EF_TESTS_DIR):
 
 # Ensures the commands for $(EF_TESTS_DIR) always run on `make setup`, regardless if the directory exists
 .PHONY: $(EF_TESTS_DIR)
-setup: $(EF_TESTS_DIR) 
+setup: $(EF_TESTS_DIR)
 
-setup-kakarot: pull-kakarot	
+setup-kakarot: pull-kakarot
 	cd lib/kakarot && make setup && make build
 
 pull-kakarot:
 	git submodule update --init --recursive
 
-# Runs the Ethereum Foundation tests
-ef-tests:
-	cargo nextest run -p ef-testing --features ef-tests 
+# Runs the repo tests
+tests:
+	cargo nextest run --all-features
 
-# Runs specific test
+# Runs ef tests only
 ef-test:
-	TARGET=$(target) cargo test -p ef-testing --features ef-tests -- --nocapture
+	cargo nextest run --package ef-testing --test tests --features ef-tests -- blockchain_tests
```

Comment:
could you explain this? why are you passing the --test flag? And what is the use of `-- blockchain_tests`?

Reply:
I guess that this means only the `mod` `blockchain_tests` in the files `tests` folder. This command has been generated by VS code actually, but the doc says

```
--test [<NAME>]           Test only the specified test target
```

and https://doc.rust-lang.org/cargo/reference/cargo-targets.html#integration-tests

Reply:
I checked the docs and so this will run the integration test file `tests` (so `tests/tests.rs` or `tests/tests/mod.rs`). I am not sure what the positional argument `blockchain_tests` does, I guess it runs the tests from the `blockchain_tests` module? I think we could remove this.

Reply:
for the record, the output of the TARGET=.. make ef-test at the end is like

```
------------
     Summary [  14.902s] 51 tests run: 50 passed, 1 failed, 0 skipped
        FAIL [  13.328s] ef-testing::tests blockchain_tests::st_mem_expanding_eip150_calls
```

with failing tests generating something like

```
        FAIL [  13.328s] ef-testing::tests blockchain_tests::st_mem_expanding_eip150_calls

--- STDOUT:              ef-testing::tests blockchain_tests::st_mem_expanding_eip150_calls ---

running 1 test
2023-10-09T13:17:40.726770Z  INFO ef_testing::models::case: Running test CallAndCallcodeConsumeMoreGasThenTransactionHasWithMemExpandingCalls_d0g0v0_Shanghai
2023-10-09T13:17:41.790713Z  WARN sequencer::sequencer: Transaction execution reverted: Error in the called contract (0x01ee6133613c44c36bd4c2808e9ac2079b18cc9e6dd54d54c7e684175b32fe8b):
Error at pc=0:7:
Got an exception while executing a hint: Hint Error: Error in the called contract (0x01ee6133613c44c36bd4c2808e9ac2079b18cc9e6dd54d54c7e684175b32fe8b):
Error at pc=0:31:
Got an exception while executing a hint: Hint Error: Error in the called contract (0x0000000000000000000000000000000000000000000000000000000000000001):
Error at pc=0:2811:
Couldn't compute operand op1. Unknown value for memory cell 36:1
Cairo traceback (most recent call last):
Unknown location (pc=0:19135)
Unknown location (pc=0:18680)
Unknown location (pc=0:18365)
Unknown location (pc=0:18281)
Unknown location (pc=0:18281)
Unknown location (pc=0:18281)
Unknown location (pc=0:18281)
Unknown location (pc=0:18281)
Unknown location (pc=0:18281)
Unknown location (pc=0:18281)
Unknown location (pc=0:18281)
Unknown location (pc=0:18281)
```

we could branch here the python utils and displays a decoded trace instead